flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -663,10 +663,16 @@ public CheckpointingMode getCheckpointingMode() {
      * org.apache.flink.runtime.state.CheckpointStorage} which configures how and where state
      * backends persist during a checkpoint.
      *
+     * @deprecated The method is marked as deprecated because starting from Flink 1.19, the usage of
+     *     all complex Java objects related to configuration, including their getter and setter
+     *     methods, should be replaced by ConfigOption. In a future major version of Flink, this
+     *     method will be removed entirely. It is recommended to switch to using the ConfigOptions
+     *     provided for configuring state backend.
```

Comment:
I would suggest adding sample code to help users know how to use ConfigOption instead

Reply:
Great idea, I'll add some sample code to demonstrate how to use ConfigOption for better clarity, and I've also included links to the corresponding documentation in the JavaDoc. And I will open a new PR soon to further update the documentation based on these changes. Thank you for the suggestion.